### ISSUE TITLE
API subscription expired

### ISSUE THREAD

**Issue by PrimaveraNL** [0x5e/homebridge-tuya-platform#66]
Not a bug of the plugin BUT
after the trial month I get the error
API subscription expired. Please renew the API subscription at Tuya IoT Platform
and all devices stop responding in homekit although they are still visible.

I remember seeing the option for 6 month renewal but I cannot find it on Tuya IOT platform any more.
Has anyone found a solution for this?

**Comment by 0x5e** [comment]
I'm not sure where to submit trial request too, but @ipa64 succeed.
https://github.com/0x5e/homebridge-tuya-platform/issues/14#issuecomment-1296375297

**Comment by PrimaveraNL** [comment]
Thx.
I found it in the mean time, very obscure.
In any case; when the API subscription is expired, the plugin stops working. 
Making this remark for documentation purposes, I understand this is out of your field of influence.
Thank you for the effort you put in this plugin though.

**Comment by 0x5e** [comment]
OK I will update a much detailed message and document later after I extend the trail month. Thanks for the understanding, it's a challenge for me to write a long and specific guide in english :)

**Comment by hattabatatta** [comment]
If I may be honest: this crap with the trial period is the biggest crap there is. Nowhere in all these installation guides is it mentioned that the tool no longer works after a month because a trial period has expired, which you then have to renew manually.
Of course, you could just take the subscription plan ... the cheapest version ONLY costs $25,000 anyway. We've all just shaken it out of our hands.
I only noticed this rubbish with the trial period today when I was sitting in a freezing cold flat because none of my heaters were switched on -> Subscription expired.

I wish there was another tool that worked with my Wi-Fi radiators and Homekit so that I wouldn't have to deal with this mess every month.

**Comment by 0x5e** [comment]
@hattabatatta You can extend 6 months at once, not every month. Of course the price is unresonable for individual users, but there's nothing I can do with the public API and SDK.
Lan protocol is private and not published, you can find the local plugins with reverse engineering, but may not working on all firmwares.

**Comment by hattabatatta** [comment]
Yes, I saw the possibility of extending it by 6 months.
But that doesn't make the whole thing any less absurd imho.
I have already considered reverse engineering the apps of the providers, but unfortunately I only have iOS and dealing with it is a horror for me as a pure .NET developer :D

**Comment by 0x5e** [comment]
Yes, "extending every 6 months" is the tuya company's decision, I can do nothing with that.
I'm able to implement the lan protocol but I'm not allowed to do this, I think.
You can search "tuya local" keywords on github.

**Comment by hattabatatta** [comment]
Don't worry I didn't mean it as an accusation ;)
I'll see if I want to read up on the subject and then possibly create my own solution. The trial period is really totally stupid.

**Comment by PrimaveraNL** [comment]
The lan solution would be a possibility maybe, but you need to obtain the local key for the device first and I think that is only possible through the api? Which would mean that communication can be done locally, but you need a way of getting to the local key.

**Comment by 0x5e** [comment]
Yes I agree :) @hattabatatta 
Yes localkey can only get from api, but it won't changed unless the device got removed/unpaired. @PrimaveraNL

**Comment by PrimaveraNL** [comment]
This is true, but when you would want to add a new device, you need a key for that. 
Unless you set it up and then do not expand your smart home, you will need an account with Tuya after the trial period.

BTW I got my whole realm working again after the subscription was renewed. There is no info on what the policy is after a 6 month trial.

**Comment by 0x5e** [comment]
I think the first-time trial period won't get removed, so there's always a way to get the keys, create an new account is free, just cost time.
But the extend period policy indeed has the possibility to change, it depends.

**Comment by Senna46** [comment]
I have extended my IoT core for 6 months. However, the error continues.
```
[TuyaOpenAPI] API subscription expired. Please renew the API subscription at Tuya IoT Platform.
Fetching home list failed. code=28841002, msg=No permissions. Your subscription to cloud development plan has expired.
```
How long does it take to reconnect? Quite inconvenient.

**Comment by PrimaveraNL** [comment]
The subscription needs to be 'reviewed', I think it took about 1,5 days in my case.
Maybe over the weekend it will take longer?

**Comment by Senna46** [comment]
Review approved, deadline extended.
But it cannot make API request.

**Comment by 0x5e** [comment]
> Review approved, deadline extended.
> 
> But it cannot make API request.

From your description seems like a cloud side bug, please submit a ticket in the tuya platform.

**Comment by 0x5e** [comment]
https://github.com/tuya/tuya-homebridge/issues/321

**Comment by Senna46** [comment]
After submitting the ticket, my issue was resolved.
thx

**Comment by four43** [comment]
This whole Tuya platform seems like classic Chinese junk - super cheap poorly made hardware, 3rd rate connection platform that we have to work around because they want to spy on the data, not let us control it locally. I know my "social credit score" will drop for saying that but I don't really care.

**Comment by githubnavigator** [comment]
@0x5e, thanks for sharing your perspective on the privacy. It's encouraging to hear. Do we need anything other than the free IoT Core subscription? I ask because when I log into my Homebridge Service API tab on the IoT Platform, I see these services enabled:

1. IoT Core
2. Smart Home Basic Service
3. Device Status Notification
4. Industry Basic Service

Are those last three included in the IoT Core? And, if not, are they free?

Thanks.

**Comment by 0x5e** [comment]
@githubnavigator  It's written on https://github.com/0x5e/homebridge-tuya-platform#configuration, it depends on what type of device you have. they are all free, but need to extend trial every 6 month :(
